### DISCUSSION TITLE
Setting that makes it so gameplay UI cannot be seen by the player but can be captured by streaming programs.

### DISCUSSION THREAD

**Discussion by GOINGCRAZYWITHIT** [ppy/osu#24688] [Ideas]
Rather this is done through a seperate window, some magic where the UI is only visible in a game capture, or some other method, this would be a first-party solution to display map and performance information.

**Comment by peppy** [comment]
This is pretty high effort and would require some very serious framework changes. Don't get your hopes up (maybe just get used to playing with the UI visible?)

**Comment by ItsShamed** [comment]
I think a better alternative would be something that #18129 would achieve, that is: broadcasting the game state to external software for use in streaming overlays (tldr: gosumemory but in-game).

That would effectively unclutter the in-game HUD for the player while providing the same (if not more) amount of information and customisation — that the skin editor provides — to the viewers.

**Reply by peppy** [reply]
You're proposing something completely different. The original user specifically asked for first party display support.

**Reply by ItsShamed** [reply]
Well, in any case, this kind of first-party support would require so much effort that you could consider that this will not happen at all.
So I just proposed a probably more feasible alternative solution to their intent – that is, separating gameplay UI and streaming UI – even if this involves third-party software to some extent.
People have been fine with using third-party tools for osu! streaming for ages, so I don't get why this would be an issue now.

**Reply by GOINGCRAZYWITHIT** [reply]
This is a nice side option I agree, and it's not that using third party tools is an issue, it's that with how much QoL additions and changes have been made to lazer it almost feels weird to be using extra tools that have to be added and individually moved and sized in OBS when all of that can now be done in the skin editor, besides I guess the map's picture, which was already easily showable with like 80 - 90% dim, but now with background blur in the mix as well it's easier still.

Having everything centralized would be ideal, but obviously it takes time and effort.
